File: example/Phenology_Model/Python/CalendarModule.py
import Calendar
import Phase
def CalculateCalendar(phase, currentdate, cumulTT,calendar):
    if (Calendar.IsMomentRegistred(Phase.PreviousMoment(phase), calendar) == 0):
        calendar = Calendar.CalendarSet(Phase.PreviousMoment(phase), currentdate, cumulTT, calendar)
    return calendar

File: example/Phenology_Model/C#/CalendarModule.cs
using System;
using System.Collections.Generic;

public static class CalendarModule
{
	public static void CalculateCalendar(double phase,  DateTime currentdate, double cumulTT, Dictionary <GrowthStage, Dictionary<DateTime, double>> calendar)
	{
    	if (Calendar.IsMomentRegistred(Phase.PreviousMoment(phase), calendar) == 0)
    		calendar = Calendar.CalendarSet(Phase.PreviousMoment(phase), currentdate, cumulTT, calendar);
	}
}
